docs/developer-docs/functionality/http_requests/http_requests-how-it-works.md

```diff
@@ -0,0 +1,152 @@
+
+To summarize, by running the responses received by the replicas through consensus, we ensure that every replica provides the same response to the smart contract Wasm execution environment, and the optional transformation function makes sure that the response can make it through consensus by removing differences.
+
+Figure goes here
+
+The canister HTTP requests feature is implemented as part of the replica and is exposed as an API on the management canister. We next outline how a request is processed by the system. The HTTP request functionality is realized at the level of subnets, i.e., each subnet handles canister HTTP requests of its canisters independently of other subnets.
+* A canister makes an HTTP outcall request with the management canister API `http_request`.
+* The request is stored temporarily in replicated state of the subnet.
+* Periodically (each round) an *adapter* at the networking layer in each replica fetches the pending HTTP outcalls from replicated state. (In fact, a 'shim' layer of the adapter that is inside the replica process does so as the adapter itself is sandboxed into a separate OS-level process for security reasons.)
+* The adapter on each replica executes the HTTP request by sending it to the target server.
+* The HTTP response from the server is received by the adapter on each replica of the subnet. The adapter limits the response size to `max_response_bytes` which defaults to $2$ MB and can be set to lower values. It is important to set this as low as reasonably possible for the expected response as it affects the price of the request.
+* An optional transformation function implemented as part of the canister is invoked on the response on each replica to transform it.
+* The transformed response is handed over to consensus on each replica.
+* Consensus agrees on a response if at least $2/3$ of the replicas have the same response for the request as input. In this case, consensus provides this response back to the management canister API at the execution layer, or an error if no consensus can be reached or in case of other problems.
+* The execution layer provides the response or error back to the calling canister.
```

Comment:
both bullet points talk about returning the result, feels al lot like duplication. Also, `consensus provides this response back to the management canister API at the execution layer` is _really_ hard to understand, especially to an outsider